library/core/src/intrinsics.rs

```diff
@@ -1724,6 +1724,25 @@ extern "rust-intrinsic" {
     #[rustc_nounwind]
     pub fn fmaf64(a: f64, b: f64, c: f64) -> f64;
 
+    /// Returns `a * b + c` for `f32` values.
+    ///
+    /// The operation is fused if the code generator determines that target
+    /// instruction set has support for a fused operation, and that the fused
+    /// operation is more efficient than the equivalent, separate pair of mul
+    /// and add instructions.
+    #[rustc_nounwind]
+    #[cfg(not(bootstrap))]
+    pub fn fmuladdf32(a: f32, b: f32, c: f32) -> f32;
```

Comment:
Is this a good opportunity to use our fallback bodies, with a fallback implementation of `a* b + c`? Const-eval should also use that.

Miri however should probably roll some dice to make both behaviors observable.

Reply:
I don't have strong rationale for choosing the fused versus separate, but it was the simplest to implement and presumably to change if/when Cranelift adds a corresponding intrinsic.

Could you give a pointer for how/where to implement the "make both behaviors observable" in Miri?

Reply:
The miri implementation would go in `src/tools/miri/src/intrinsics/mod.rs`. There you can use `this.machine.rng.borrow_mut()` to get access to an RNG that you can use to decide whether to use the fused implementation or the double-rounding implementation. That way, if the function is called many times in Miri, both behaviors will be observable.

Reply:
I implemented this and added a test in the spirit of `is_val_statically_known` in `src/tools/miri/tests/pass/intrinsics/intrinsics.rs` (looping 50 times to observe both outcomes). Let me know if there is a better way to ensure that.